acme_test.go

```diff
@@ -155,6 +159,81 @@ func TestAuthorize(t *testing.T) {
 	}
 }
 
+func TestAcceptChallenge(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method == "HEAD" {
+			w.Header().Set("replay-nonce", "test-nonce")
+			return
+		}
+		if r.Method != "POST" {
+			t.Errorf("r.Method = %q; want POST", r.Method)
+		}
+
+		// Decode request
+		body, err := ioutil.ReadAll(r.Body)
+		if err != nil {
+			t.Fatal(err)
+		}
+		var bodyJSON map[string]string
+		err = json.Unmarshal(body, &bodyJSON)
+		payload, err := base64.RawURLEncoding.DecodeString(bodyJSON["payload"])
+		if err != nil {
+			t.Fatal(err)
+		}
+		var j map[string]string
```

Comment:
It is normally not a good practice to use a generic `map[string]string` type, it doesn't give you any advantage here. A safer way is to use a more strict type:

    var j struct {Resource, Type, KeyAuthorization string}